packages/gatsby/src/utils/page-data.ts

```diff
@@ -173,11 +173,11 @@ export async function flush(): Promise<void> {
   return
 }
 
-export function enqueueFlush(): void {
+export async function enqueueFlush(): Promise<void> {
```

Comment:
Are you sure this is going to work as intended?

`gatsby/packages/gatsby/src/services/start-webpack-server.ts` uses this inside an async function, is not updated in this PR to wait for this promise
`gatsby/packages/gatsby/src/query/index.js` uses this (as `pageDataUtil.enqueueFlush()`), is NOT async, and is not updated in this PR to take this change into account.
`gatsby/packages/gatsby/src/state-machines/query-running/actions.ts` uses this in a function that is NOT async. That's the file with the empty line, did you update the usage there initially?

Reply:
@sidharthachatterjee

Reply:
Yup, this is intended. Apologies for the confusion. Let me explain. 

The `enqueueFlush` exists along with the `flush` fn. The latter always returned a promise. And so far (before this PR), `enqueueFlush` was used as a fire and forget everywhere, where we'd want a flush to occur but wouldn't necessarily want to wait for it to complete. 

I assumed that we'd need to be able to wait for an "enqueued flush" in the case of the first usage that you listed (start-webpack-server)

Turns out we don't because the state machine is now coordinating these states. I'll remove this change as a result. On it.